packages/select/src/common/listItemsProps.ts

```diff
@@ -124,6 +124,22 @@ export interface IListItemsProps<T> extends IProps {
      */
     onQueryChange?: (query: string, event?: React.ChangeEvent<HTMLInputElement>) => void;
 
+    /**
+     * If provided, allows new items to be created with the current query string.
+     * This is invoked when user interaction causes a new item to be created, either by pressing the `enter` key or
+     * by clicking on the "Create Item" option. It transforms a query string into an item type.
+     */
```

Comment:
nit: Suggested edits for clarity:

```tsx
/**
 * If provided, allows new items to be created using the current query
 * string; it should transform the query string into a new item. Will be
 * invoked when the user selects the item returned from
 * `createNewItemRenderer` (either via `Enter` or click).
 */
```

Also, should we require `createNewItemRenderer` to be defined if this callback is defined (and vice versa)?